### PR TITLE
Format toolbar: visual clue for hidden active items

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

clue

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jasmussen** [comment]
My goodness, this feels like an excellent win!

And serendipitously, I just discussed the other day on another ticket with @karmatosed exactly this treatment — what does a toolbar button look like when it has opened a submenu?

In this case, obviously, the black "toggled" state does not indicate simply that a popover is open, but simply that a submenu item was toggled. I imagine this would benefit alignments too, do you think?

But the remaining question is: what do we do with the indication that a submenu is open? Right now there's just a blue color:

Screenshot 2020-04-27 at 09 48 07

Some options:

- We keep the behavior as is — if it's a toggled submenu then the treatment is as it is in this branch.
- We redesign the "open submenu" treatment in some way that works both for menus without toggles, and for menus with — blue on black won't be sufficient contrast.
- We remove the blue entirely — do we need submenu-is-open indication?

**Comment by ellatrix** [comment]
Yeah, do we need a visual clue for a submenu being open? In my opinion, the submenu appearing is enough of a visual clue.

**Comment by ellatrix** [comment]
But I like extending this active visual clue to alignments, it makes sense.

**Comment by ellatrix** [comment]
@jasmussen I added it for text alignment too.

**Comment by jasmussen** [comment]
> I added it for text alignment too.

Cool!

How about block alignments?

**Comment by karmatosed** [comment]
Love this #21814 is the issue where I brought up the select state and maybe we can combine with this to close?

**Comment by youknowriad** [comment]
Do we still need the "text color" button to show up separately with this change?

**Comment by ellatrix** [comment]
@youknowriad What do you mean? :)

**Comment by jasmussen** [comment]
If you select text, change it's color using the button in the drop-down, that color button now shows up among the other inline buttons instead of in the drop-down.

With this visual indication, perhaps it can just stay in the drop-down.

**Comment by ellatrix** [comment]
@jasmussen @karmatosed How's this? :)

**Comment by jasmussen** [comment]
It seems good to me:

inline

The missing text color is my theme fault, not the code here. 

Did you want to move the inline text color indication as Riad suggested? Or is that a separate PR?

**Comment by ellatrix** [comment]
Wasn't it but in the block toolbar for a reason? I didn't work on the text color button, so I don't know why it appears in the block toolbar when it's active. I assume so it's easy for people to know about the color?

**Comment by ellatrix** [comment]
Aaah, or do you mean, now that the collapsed button indicates active state, we can move it back?

**Comment by jasmussen** [comment]
> Aaah, or do you mean, now that the collapsed button indicates active state, we can move it back?

Yes, essentially, because there's indication as to where you go to change it. There were also some challenges with the color button in the toolbar.

**Comment by ellatrix** [comment]
Screenshot 2020-05-04 at 11 51 51

I moved the active button, which is much better and more consistent imo. The color indicator is in the wrong position though. Here again I wonder... why is there a color indicator? You can see the color in both the text AND the color selection UI.

**Comment by ellatrix** [comment]
Ok, I removed it. Ready for review.

**Comment by jasmussen** [comment]
I think there is some value to showing the color swatch for the inline color. It's sort of a way to "augment the icon". I was just about to fix it, actually. `components-dropdown-menu__menu-item` just needs position relative, and `format-library-text-color-button__indicator` needs `bottom: 8px` instead of 10px.

**Comment by jasmussen** [comment]
Screenshot 2020-05-04 at 12 03 35

**Comment by mapk** [comment]
I just tested this PR in Gutenberg.run and noticed a few issues.

### 1. Color indicator missing.
The color bar is missing from the "A" icon. I should see a red bar in the screenshot below.

Screen Shot 2020-05-04 at 4 57 27 PM

### 2. Off-centered icon
The dropdown arrow appears off-centered in the selected state of the icon.

Screen Shot 2020-05-04 at 4 58 49 PM

### 3. Pressed state loses the icon
When I press down on the dropdown icon, the icon disappears. Perhaps it is just displaying in black on a black background.

Screen Shot 2020-05-04 at 5 00 08 PM

Tested on Chrome 81.0.4.

**Comment by jasmussen** [comment]
> 2. Off-centered icon
> The dropdown arrow appears off-centered in the selected state of the icon.

Eagle eye, Mark! This is actually not a regression of this PR but is present in master:

Screenshot 2020-05-05 at 09 03 02

But definitely good to fix of course.

**Comment by ellatrix** [comment]
> The color bar is missing from the "A" icon. I should see a red bar in the screenshot below.

@mapk I removed the color indication because it doesn't make any sense to me there. The icons in the list should be consistent and there's color indication in both the text and the color panel that it opens.

**Comment by ellatrix** [comment]
> The dropdown arrow appears off-centered in the selected state of the icon.

I'm not sure how to fix this. Is the icon in the SVG not centred?

**Comment by jasmussen** [comment]
I have to agree with Mark on this one — the icons are merely indicators of the function of the tool underneath, and the changing color helps convey that. It's also widely used:

color

One thing that could help consistency, would be to always show that colored bar below the A, but just leave it uncolored until the user picks a color. 

I'll look at centering the dropdown separately, as it's not a regression of this PR!

**Comment by ellatrix** [comment]
Ok, I'll restore the color indicator.

I looked at the active state, but I can't figure out why that particular button is affected and all the other buttons not.

**Comment by jasmussen** [comment]
The uncentered chevron is a fun one, it's a rounding issue:

Kapture 2020-05-05 at 11 03 03

I'll keep thinking about this and how to fix it, but just to reiterate, it's not a regression of this branch.

File: packages/components/src/toolbar/style.scss [hunk 1]
```diff
@@ -62,7 +62,7 @@
 				background: transparent;
 			}
 
-			&::before {
+			&:not(:active)::before {
```
Comment (ellatrix) [inline packages/components/src/toolbar/style.scss:65]: @jasmussen Is this the right way? I'm not sure :)

Comment (jasmussen) [reply packages/components/src/toolbar/style.scss:65]: I can't immediately grok what it does. But my instinct was to simply add a separate :active modifier and make the text white there. But I can see that the following CSS would interfere with that:

```
.components-button:not([aria-disabled="true"]):active {
    color: inherit;
}
```

So if what you have here works, I think it's good to ship. I don't know if it needs a comment.
Nice work.

**Comment by ellatrix** [comment]
Is this good to go?

**Comment by jasmussen** [comment]
If the color indicator is back yes I think it's good 💯

**Review by mapk — APPROVED** [review]
Looks great!

visual

**Comment by aristath** [comment]
Is this one still relevant? If yes, then it needs to be rebased :+1:

**Comment by ramonjd** [comment]
🎉 Agree that this one would be a nice win and great to land. Looks like it's ready to merge with a rebase @ellatrix?

**Comment by Mamaduka** [comment]
I agree with everyone. This would be a sweet improvement to land.

**Comment by mtias** [comment]
Nice to see the closed loop :)

**Pull request merged**